File: mrr-upload/src/features/foodlog/entities/food.ts
import Joi from "joi";

export enum FoodSource {
  USER_DB = "user_db",
  SYSTEM_DB = "system_db",
}

export interface Food {
  /** The id from the source database */
  id: string;
  /** The main name of the food */
  foodName: string;
  /** Other possible names that the food is known as */
  otherNames: string[];
  /** Attributes that describe the food, such as cooked, raw, etc */
  description: string[];
  /** The amount of food to compare the nutritional values */
  portionSize: number;
  /** The unit of the portion size */
  portionSizeUnit: string;
  /** The recommended amount of food to be consumed */
  servingSize: number;
  /** The unit of the serving size */
  servingSizeUnit: string;
  /** The source of the food, either from the user database or the system database */
  foodSource: FoodSource;
  /** The amount of calories in grams */
  calories: number;
  /** The amount of protein in grams */
  protein: number;
  /** The amount of fat in grams */
  fat: number;
  /** The amount of carbohydrates in grams */
  carbohydrates: number;
}

export const FoodCreateInputSchema = Joi.object({
  id: Joi.string().required(),
  foodName: Joi.string().required(),
  otherNames: Joi.array().items(Joi.string()).required(),
  description: Joi.array().items(Joi.string()).required(),
  portionSize: Joi.number().positive().required(),
  portionSizeUnit: Joi.string().required(),
  servingSize: Joi.number().positive().required(),
  servingSizeUnit: Joi.string().required(),
  calories: Joi.number().min(0).required(),
  protein: Joi.number().min(0).required(),
  fat: Joi.number().min(0).required(),
  carbohydrates: Joi.number().min(0).required(),
  foodSource: Joi.string()
    .valid(FoodSource.USER_DB, FoodSource.SYSTEM_DB)
    .required(),
}).meta({ className: "FoodCreateInput" });

File: speech2nutrition/core/domain/entities/food.py
from enum import Enum
from typing import List

from pydantic import BaseModel


class FoodSource(str, Enum):
    user_db = "user_db"
    system_db = "system_db"


# Note all macro nutrients are in grams
class Food(BaseModel):
    id: str
    """ The id from the source database """
    food_name: str
    """ The main name of the food """
    other_names: List[str]
    """ Other possible names that the food is known as """
    description: List[str]
    """ Attributes that describes the food, such cooked, raw, etc """
    full_description: List[str]
    """ The full description of the food, including the food name and the description. Each item in the list is is clean using NLP techniques"""
    portion_size: float
    """ For USDA Foundation food reference units are in grams"""
    portion_size_unit: str
    """ The unit of the portion reference """
    serving_size: float
    """ The recommended amount of food to be consumed """
    serving_size_unit: str
    """ The unit of the serving size """
    food_source: FoodSource
    """ The source of the food, either from the user database or the system database """
    calories: float
    protein: float
    fat: float
    carbohydrates: float
